server/conn_stmt.go

```diff
@@ -280,6 +285,8 @@ func (cc *clientConn) handleStmtFetch(ctx context.Context, data []byte) (err err
 		prepareObj, _ := cc.preparedStmtID2CachePreparedStmt(stmtID)
 		if prepareObj != nil && prepareObj.SQLDigest != nil {
 			ctx = topsql.AttachSQLInfo(ctx, prepareObj.NormalizedSQL, prepareObj.SQLDigest, "", nil, false)
+			sessVars.StmtStats.OnSQLDigestReady(prepareObj.SQLDigest.Bytes())
```

Comment:
It seems that `OnSQLDigestReady` will be called multiple times in the same execution. Is this expected? If so, I suggest to describe it in the comment of `StatementObserver.OnSQLDigestReady` that this method may be called multiple times and must be no side effects within a single execution.

Reply:
Already add some comment. `OnSQLDigestReady` is ok to call multiple times.
But currently, `OnSQLDigestReady` won't call multiple times for 1 statement. here is a special handle for `stmt fetch` statement.